src/SFML/System/Win32/LibraryLoader.hpp

```diff
@@ -0,0 +1,120 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_LIBRARY_LOADERWIN32_HPP
+#define SFML_LIBRARY_LOADERWIN32_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <SFML/System/NonCopyable.hpp>
+#include <SFML/System/String.hpp>
+#include <windows.h>
```

Comment:
I don't like exposing `windows.h` in public headers. I would suggest changing this class to use the PImpl idiom, unless the intention for `LibraryLoader.hpp` is to be an implementation header, and only expose it through PImpl to the users. Please confirm.

Reply:
Also, do we want
```cpp
#ifndef WIN32_LEAN_AND_MEAN
#define WIN32_LEAN_AND_MEAN
#endif
```
here?

Reply:
I feel the same here for both sentenses.

Since we decided to move LibraryLoader from internal implementation to public, I'll implement PImpl the same way other classes in SFML are implemented. This should deal with exposing windows.h.

The thing with WIN32_LEAN_AND_MEAN - it's better to add but we are using it in some files and not using in other.
This is confusing. What do you think if we will add a small header file for including windows.h? I've done an implementation for another project, you can see it here https://github.com/teonet-co/teobase/blob/master/include/teobase/windows.h